pkg/model/components/nodeterminationhandler.go

```diff
@@ -39,12 +39,19 @@ func (b *NodeTerminationHandlerOptionsBuilder) BuildOptions(o *kops.Cluster) err
 		clusterSpec.CloudProvider.AWS.NodeTerminationHandler = &kops.NodeTerminationHandlerSpec{}
 	}
 	nth := clusterSpec.CloudProvider.AWS.NodeTerminationHandler
+	if nth.Enabled == nil {
+		if clusterSpec.Karpenter != nil && clusterSpec.Karpenter.Enabled {
+			nth.Enabled = fi.PtrTo(false)
```

Comment:
This could use a comment IMO...

Reply:
should we check this in api validation? to ensure `spec.karpenter.enabled != spec.cloudprovider.aws.nodeTerminationHandler.enabled` ?